packages/core/src/components/TreeView/TreeView.js

```diff
@@ -5,33 +5,68 @@ import { noop } from '../../common/helpers';
 
 import TreeViewNode from './TreeViewNode';
 
-const TreeView = ({
-  nodes,
-  selectNode,
-  highlightOnSelect,
-  highlightOnHover
-}) => {
-  const classes = classNames('list-group', {
-    'treeview-pf-select': highlightOnSelect,
-    'treeview-pf-hover': highlightOnHover
-  });
-
-  return (
-    <div className="treeview">
-      <ul className={classes}>
-        {nodes &&
-          nodes.map((node, index) => (
-            <TreeViewNode
-              node={node}
-              key={index}
-              level={1}
-              selectNode={selectNode}
-            />
-          ))}
-      </ul>
-    </div>
-  );
-};
+class TreeView extends React.Component {
+  state = {
+    focusedNodeId: ''
+  };
+
+  onFocus = node => {
+    this.setState(() => ({ focusedNodeId: node.id }));
+  };
+
+  onKeyDown = event => {
+    const nodes = this.getVisibleNodes([...this.treeRef.current.children]);
+    const currentNodePosition = nodes.findIndex(
+      element => element.id === this.state.focusedNodeId
+    );
+    if (event.key === 'ArrowDown' && currentNodePosition !== nodes.length - 1) {
+      nodes[currentNodePosition + 1].focus();
+    } else if (event.key === 'ArrowUp' && currentNodePosition !== 0) {
+      nodes[currentNodePosition - 1].focus();
+    }
+  };
+
+  getVisibleNodes = nodes =>
+    nodes.filter(node => !node.className.match(/node-hidden/));
+
+  treeRef = React.createRef();
+
+  render() {
+    const {
+      nodes,
+      selectNode,
+      highlightOnHover,
+      highlightOnSelect
+    } = this.props;
+    const classes = classNames('list-group', {
+      'treeview-pf-select': highlightOnSelect,
+      'treeview-pf-hover': highlightOnHover
+    });
+
+    return (
+      <div className="treeview">
```

Comment:
I've seen some [successful examples](http://www.oaa-accessibility.org/examplep/treeview1/) using `role="application"` on an outer element like this. Since we're hi-jacking the keyboard flow once focus is within the tree, it in some ways fits this description nicely. I'm curious how this might affect various screen reader experience.

https://www.w3.org/TR/wai-aria-1.1/#application

It comes into play when talking about [managing focus](https://www.w3.org/TR/wai-aria-1.1/#managingfocus).